core/src/main/java/com/sequenceiq/cloudbreak/service/publicendpoint/FreeIPAEndpointManagementService.java

```diff
@@ -43,10 +43,13 @@ public void registerLoadBalancerDomainWithFreeIPA(Stack stack) {
             try {
                 if (StringUtils.isNotEmpty(loadBalancer.getDns())) {
                     sendAddDnsCnameRecordRequest(stack, loadBalancer);
-                } else if (StringUtils.isNotEmpty(loadBalancer.getIp())) {
+                } else {
+                    LOGGER.error("Unable to find DNS for load balancer. Load balancer address will not be registered as CNAME with FreeIPA.");
+                }
+                if (StringUtils.isNotEmpty(loadBalancer.getIp())) {
                     sendAddDnsARecordRequest(stack, loadBalancer);
```

Comment:
After I moved to private load-balancers for public networks in Azure I am getting this error [1]. While technically I should not because the IP address is added as an A record.

[1]
```
/opt/cloudera/cm/bin/gen_credentials_ipa.sh failed with exit code 1 and output of <<
+ CMF_REALM=GK2DEV.XCU2-8Y8X.DEV.CLDR.WORK
+ export PATH=/usr/kerberos/bin:/usr/kerberos/sbin:/usr/lib/mit/sbin:/usr/sbin:/usr/lib/mit/bin:/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
+ PATH=/usr/kerberos/bin:/usr/kerberos/sbin:/usr/lib/mit/sbin:/usr/sbin:/usr/lib/mit/bin:/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
+ kinit -k -t /etc/cloudera-scm-server/cmf.keytab CM/test9-manager0.gk2dev.xcu2-8y8x.dev.cldr.work@GK2DEV.XCU2-8Y8X.DEV.CLDR.WORK
+ KEYTAB_OUT=/var/run/cloudera-scm-server/cmf2646773110601411.keytab
+ PRINCIPAL=HTTP/10.2.0.8@GK2DEV.XCU2-8Y8X.DEV.CLDR.WORK
+ MAX_RENEW_LIFE=0
+ '[' -z /etc/krb5.conf ']'
+ echo 'Using custom config path '\''/etc/krb5.conf'\'', contents below:'
+ cat /etc/krb5.conf
+ PRINC=HTTP
++ echo HTTP/10.2.0.8@GK2DEV.XCU2-8Y8X.DEV.CLDR.WORK
++ cut -d / -f 2
++ cut -d @ -f 1
+ HOST=10.2.0.8
+ set +e
+ ipa host-show 10.2.0.8
ipa: ERROR: 10.2.0.8: host not found
+ ERR=2
+ set -e
+ [[ 2 -eq 0 ]]
+ echo 'Adding new host: 10.2.0.8'
+ ipa host-add 10.2.0.8
ipa: ERROR: Host '10.2.0.8' does not have corresponding DNS A/AAAA record
```

Reply:
This is fixed by preferring the FQDN for internal load balancers.